esp8266/uart.c

```diff
@@ -200,6 +200,20 @@ bool uart_rx_wait(uint32_t timeout_us) {
     }
 }
 
+int uart_rx_any(uint8 uart) {
+    if (input_buf.iget != input_buf.iput) {
+        return true; // have at least 1 char ready for reading
+    }
+    return false;
+}
+int uart_tx_any(uint8 uart) {
```

Comment:
I think the logic/naming/use of this function is wrong.  It returns true if there are characters waiting to be transmitted.  Instead we want a function that tells us if there is room in the tx buffer.